jiant/tasks/lib/qasrl.py

```diff
@@ -0,0 +1,81 @@
+import gzip
+import nltk
+import json
+
+from jiant.tasks.lib.templates import span_prediction as span_pred_template
+from jiant.utils.retokenize import TokenAligner
+
+
+class QASRLTask(span_pred_template.AbstractSpanPredicationTask):
+    def get_train_examples(self):
+        return self._create_examples(self.train_path, set_type="train")
+
+    def get_val_examples(self):
+        return self._create_examples(self.val_path, set_type="val")
+
+    def get_test_examples(self):
+        return self._create_examples(self.test_path, set_type="test")
+
+    def _create_examples(self, file_path, set_type):
+
+        with gzip.open(file_path) as f:
+            lines = f.read().splitlines()
+
+        examples = []
+        ptb_detokenizer = nltk.tokenize.treebank.TreebankWordDetokenizer()
+
+        for line in lines:
+            datum = json.loads(line)
+            datum = {
+                "sentence_tokens": datum["sentenceTokens"],
+                "entries": [
+                    {
+                        "verb": verb_entry["verbInflectedForms"]["stem"],
+                        "verb_idx": verb_idx,
+                        "questions": {
+                            question: [
+                                [
+                                    {
+                                        "tokens": datum["sentenceTokens"][span[0] : span[1] + 1],
+                                        "span": span,
```

Comment:
@zphang recommended this to me in an [earlier review](https://github.com/jiant-dev/jiant/pull/98#discussion_r449916762): use " Inclusive/Exclusive span type" to avoid confusion.

Reply:
Ah, wait, what? This is heading towards AllenNLP preprocessing. If we push this concept further, it basically become field in AllenNLP. Then, if there are multiple input sequences, should we specify which sequence a span is defined on?  Then you have the span field of AllenNLP.

Reply:
Personally, I think enforcing close, or left close right open span at all the example classes will a cleaner solution. After all, spans will be processed by either a span input module or a span prediction module, they all need to match with the interface of those modules.